### ISSUE TITLE
array of object for body param

### ISSUE THREAD

**Issue by GiladShoham** [swagger-api/swagger-ui#1718]
Hi,
I have tried to add body param that get an array of my objects.
(something like the body in http://petstore.swagger.io/#/ -  POST /user/createWithArray)
no matter what i did, i always get in ui array of undefined in the scheme.
i have tried to remove the ref and make it very simple:

``` yaml
- in: body
        description: Array of something
        name: myArr
        required: true
        schema:
          type: array
          items:
            type: object
            properties:
              myProp:
                type: string
```

if i change it to be (as array of string instead of some object) it works fine:

``` yaml
- in: body
        description: Array of something
        name: myArr
        required: true
        schema:
          type: array
          items:
            type: string
```

im not sure if this is a swagger-ui issue or general issue.
i have also tries with array of boolean (didn't work) array of integer (worked), $ref to my scheme but any time it gets to an object it doesn't work.

any idea will be great.
Thx.

**Comment by GiladShoham** [comment]
ok,
i found very strange behavior, when i change the items to be $ref for the same object is working, i mean:
```yaml
- in: body
        description: Array of something
        name: myArr
        required: true
        schema:
          $ref: '#/definitions/myObj'
```
and my obj (under definition) is something like:
```yaml
myObj:
  type: object
  properties:
    myProp:
      type: string
```
is working.
but this should be exactly the same.

This behaviour make it very hard for me because i use my code (taken from here:
http://azimi.me/2015/07/16/split-swagger-into-smaller-files.html)
to resolve all the $ref to be one json instead of let the swagger itself to do it.
so now i somehow need to change it to not resolve only in specific case:
when it's array under parameters under body type param

**Comment by fehguy** [comment]
This has been an issue with swagger-js

But regarding the link from azimi.me--bless his heart but what he's describing simply isn't legal.  So please don't use that as a guideline, it's not real nor valid swagger.

**Comment by GiladShoham** [comment]
Thx:)

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen if you experience otherwise.
